sms.py

```diff
@@ -1,24 +1,32 @@
-import dateutil.parser
 import glob
 import os
-import phonenumbers
 import re
 import time
-from datetime import datetime, timedelta
 from base64 import b64encode
-from bs4 import BeautifulSoup
+from datetime import datetime, timedelta
 from io import open  # adds emoji support
 from pathlib import Path
 from shutil import copyfileobj, move
 from tempfile import NamedTemporaryFile
 from time import strftime
 
+import dateutil.parser
+import phonenumbers
+from bs4 import BeautifulSoup
+
 sms_backup_filename = "./gvoice-all.xml"
 sms_backup_path = Path(sms_backup_filename)
 # Clear file if it already exists
 sms_backup_path.open("w").close()
 print("New file will be saved to " + sms_backup_filename)
 
+# Constant for allowed extensions
+VIDEO_EXTENSIONS = {'.3gp', '.mp4'}
+IMAGE_EXTENSIONS = {'.jpg', '.jpeg', '.png', '.gif'}
+VCARD_EXTENSION = {'.vcf'}
+ALLOWED_EXTENSIONS = IMAGE_EXTENSIONS | VCARD_EXTENSION | VIDEO_EXTENSIONS

```

Comment:
All extensions are here rather than repeated throughout the code.
Makes it much easier to add an extension. I went through several rounds of tests because I kept missing one of those instances.